fdbserver/VersionedBTree.actor.cpp

```diff
@@ -692,19 +715,21 @@ class FIFOQueue {
 			debug_printf("FIFOQueue::Cursor(%s) loadExtent\n", toString().c_str());
 			return map(queue->pager->readExtent(pageID), [=](Reference<ArenaPage> p) {
 				page = p;
-				debug_printf("FIFOQueue::Cursor(%s) loadExtent done. Page: %p\n", toString().c_str(), page->begin());
+				debug_printf("FIFOQueue::Cursor(%s) loadExtent done. Page: %p\n", toString().c_str(), page->rawData());
 				return Void();
 			});
 		}
 
 		void writePage() {
 			ASSERT(mode == WRITE);
 			debug_printf("FIFOQueue::Cursor(%s) writePage\n", toString().c_str());
-			VALGRIND_MAKE_MEM_DEFINED(raw()->begin(), offset);
-			VALGRIND_MAKE_MEM_DEFINED(raw()->begin() + offset, queue->dataBytesPerPage - raw()->endOffset);
+
+			// Mark the unused part of the item space as defined
+			VALGRIND_MAKE_MEM_DEFINED(header()->begin() + offset, header()->itemSpace - header()->endOffset);
```

Comment:
a nit: we don't #ifdef VALGRIND protection here?

Reply:
This macro is defined to nothing if valgrind is off.